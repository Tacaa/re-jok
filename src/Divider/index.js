// @flow
import * as React from 'react';
import {
  StyledDivider,
  StyledDividerContent,
} from './style';

type Props = {
  /** Override style of divider**/
  style?: Object,
  /** Add more class to  divider**/
  className?: string,
  /** Children could be anything**/
  children?: any,
  /** Set type of divider is vertical or horizontal**/
  type: 'vertical' | 'horizontal',
  /** Set divider is dashed or not.. Note: it's not work with divider has children **/
  dashed?: boolean,
  /** Set text in divider left or right**/
  textPosition?: 'left' | 'right',
}

const defaultProps = {
  dashed: false,
  type: 'horizontal'
}

class Divider extends React.Component<Props> {

  static defaultProps = defaultProps;

  render(){
    const {
      children,
      type,
      ...rest
    } = this.props;

    const hasChild = !(children === null || children === undefined || (Array.isArray(children) && children.length === 0));

    return (
      <StyledDivider hasChild={hasChild} type={type} {...rest}>
        {children && type !== "vertical" && <StyledDividerContent>{children}</StyledDividerContent>}
      </StyledDivider>
    )
  }
}

export default Divider
