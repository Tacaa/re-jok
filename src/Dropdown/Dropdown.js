// @flow
import * as React from 'react';
import {
  StyledDropdownWrapper,
  StyledDropdown
} from './style';
import ReactDOM from 'react-dom'



type Props = {
  /** Override style of Dropdown**/
  style?: Object,
  /** Add more class to dropdown**/
  className?: string,
  /** Children of dropdown**/
  children: React$Element<any>,
  /** Dropdown content could be anything**/
  content?: any,
  /** Dropdown could be 6 position. Default value is bottom left**/
  position: 'bottomLeft' | 'bottomCenter' | 'bottomRight' | 'topLeft' | 'topCenter' | 'topRight',
  /** set trigger dropdown is hover or click**/
  trigger: 'hover' | 'click',
}

type State = {
  top: number,
  left: number,
  open: boolean
}


const defaultProps = {
  position: 'bottomLeft',
  trigger: 'hover',
}



class Dropdown extends React.Component<Props, State>{

  static defaultProps = defaultProps;

  state = {
    top: 0,
    left: 0,
    open: false
  }

  refParent: {
    current: null | React$ElementRef<any>
  } = React.createRef();


  refChildren: {
    current: null | React$ElementRef<any>
  } = React.createRef();

  refDropdown: {
    current: null | React$ElementRef<any>
  } = React.createRef();

  timer: TimeoutID;



  componentDidMount(){
    const {trigger} = this.props;


    trigger === "click" && document.addEventListener('mousedown', this.handleEvent);

  }

  componentWillUnmount(){

    const {trigger} = this.props;

    trigger === "click" && document.removeEventListener('mousedown', this.handleEvent);


  }

  resetDropdown = ()=> {

    this.setState({
      top: 0,
      left: 0,
      open: false
    })
  }

  handleEvent = (e: Event)=> {

    const {trigger} = this.props;


    const wrapper = ReactDOM.findDOMNode(this.refParent.current);
    const node = ReactDOM.findDOMNode(this.refChildren.current);
    const dropdown = ReactDOM.findDOMNode(this.refDropdown.current);
    const {open} = this.state;


    if (node && node instanceof HTMLElement && dropdown && dropdown instanceof HTMLElement && wrapper && wrapper instanceof HTMLElement) {
      if (node.contains((e.target: any)) || dropdown.contains((e.target: any)) || wrapper.contains((e.target: any))) {
        const rectChild = this.getPositionElement(node);
        const rectParent = this.getPositionElement(wrapper);
        const { width, height } = this.realElement(dropdown);

        const {top, left } = this.getRealPosition(rectChild, rectParent, width, height);

        if(open) {
          if(trigger === 'hover') return;
          setTimeout(()=>{
            this.resetDropdown()
          }, 150)
        }
        else {
          this.setState({
            top,
            left,
            // width,
            // height,
            open: true
          })
        }
      } else {
        if(trigger === 'hover') return;
        this.resetDropdown();
      }


    }
  }

   realElement = (el: HTMLElement) => {
    const clone = el.cloneNode(true);

    clone.style.display = "block";
    clone.style.top = "-9999px";
    clone.style.left = "-9999px";
    if (!document.body) throw new Error("Unexpectedly missing <body>.");
    const body: HTMLElement = document.body;
    body.appendChild(clone);
    const height = clone.offsetHeight;
    const width = clone.offsetWidth;

    clone.remove();
    return {
      width,
      height
    };
  }

  getPositionElement = (el: HTMLElement) => {
    const rect = el.getBoundingClientRect();

    const height = el.offsetHeight || el.clientHeight;
    const width = el.offsetWidth || el.clientWidth;

    return {
      top: rect.top,
      left: rect.left,
      height,
      width
    }
  }

  getRealPosition = (rectChild: Object, rectParent: Object, width: number, height: number) => {

    const {position} = this.props;


    let top = 0;
    let left = 0;
    switch (position) {
      case 'bottomLeft':
        top = rectChild.top - rectParent.top + rectChild.height + 10;
        left = rectChild.left - rectParent.left;
        break;
      case 'topLeft':
        top = rectChild.top - height -rectParent.top - 10;
        left = rectChild.left - rectParent.left;
        break;
      case 'bottomRight':
        top = rectChild.top - rectParent.top + rectChild.height + 10;
        left = rectChild.left - rectParent.left - width + rectChild.width;
        break;
      case 'topRight':
        top = rectChild.top - height -rectParent.top - 10;
        left = rectChild.left - rectParent.left - width + rectChild.width;
        break;
      case 'bottomCenter':
        top = rectChild.top - rectParent.top + rectChild.height + 10;
        left = rectChild.left - rectParent.left - width/2 + rectChild.width/2;
        break;
      case 'topCenter':
        top = rectChild.top - height -rectParent.top - 10;
        left = rectChild.left - rectParent.left - width/2 + rectChild.width/2;
        break;
      default:
        return {
          top,
          left
        }

    }

    return {
      top,
      left
    }
  }

  onMouseEnter = (e: Event) => {

    const wrapper = ReactDOM.findDOMNode(this.refParent.current);
    if (wrapper && wrapper instanceof HTMLElement)
    wrapper.contains((e.target: any)) && this.timer && clearTimeout(this.timer);

    this.handleEvent(e);
  }

  onMouseLeave = () => {

    this.timer = setTimeout(()=>{

        this.resetDropdown()

    }, 300)
  }




  render(){
    const {
      children,
      content,
      trigger,
      ...rest
    } = this.props;


    const mouseEvent = {
      onMouseLeave: trigger === 'click' ? null : this.onMouseLeave,
      onMouseEnter: trigger === 'click' ? null : this.onMouseEnter,
    }

    return (
      <StyledDropdownWrapper
        {...mouseEvent}
        innerRef={this.refParent}>

          {
            React.Children.count(children) === 1
              ? <children.type

                {...children.props}
                ref={this.refChildren}/>
              : null
          }

       <StyledDropdown {...rest} innerRef={this.refDropdown} {...this.state}>
         {content}
       </StyledDropdown>
     </StyledDropdownWrapper>
    )
  }
}

export default Dropdown
