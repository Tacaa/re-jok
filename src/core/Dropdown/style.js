import styled, {css, keyframes} from 'styled-components';

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`

export const StyledDropdownWrapper = styled.div`
  position: relative;

`

export const StyledDropdown = styled.div`
     max-width: 25.6rem;
     position: absolute;
     display: ${props =>props.open ? "block" : "none"};
    ${
      props =>
      props.open && css`
       animation: ${fadeIn} .4s;
      `
    }
    top: ${props => `${props.top}px`};
    left: ${props => `${props.left}px`};
    z-index: ${({theme}) => theme.zIndex.dropDown};
    padding: 1rem 0;
    background-color: ${({theme}) => theme.background.default};
    border: ${({theme}) => theme.border.primary};
    font-size: 1.4rem;
    border-radius: .4rem;
    box-shadow: 0 .5rem 1.2rem rgba(0,0,0,0.15);

`
