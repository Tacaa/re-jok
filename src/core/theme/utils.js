import {
  defaultTheme
} from './index';


export const configTheme = (theme: Object) => {
  let jokTheme = {...defaultTheme, ...theme};
  return jokTheme;
}
