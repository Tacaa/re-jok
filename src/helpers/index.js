export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
