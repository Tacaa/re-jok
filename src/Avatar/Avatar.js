// @flow
import * as React from 'react'

import {Icon} from '../Icon';
import {StyledAvatar, StyledAvatarImage} from './style';

type Props = {
  /** Set size of avatar.The default size is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of avatar.The default shape is circle**/
  shape: 'circle' | 'square',
  /** Set incon in avatar**/
  icon?: string,
  /** Set alt attribute to image avatar**/
  imgAlt?: string,
  /** Set src attribute to image avatar**/
  imgSrc?: string,
  /** Add more class to avatar**/
  className?: string,
  /**Overrie or add style to root avatar**/
  style?: Object,
  /****/
  children?: any
}

const defaultProps = {
  size: 'default',
  shape: 'circle'
}

class Avatar extends React.Component<Props> {
  static defaultProps = defaultProps;
  render() {
    const {
      imgAlt,
      imgSrc,
      icon,
      ...rest
    } = this.props
    return (<StyledAvatar imgSrc={imgSrc} {...rest}>
      {
        imgSrc
          ? (<StyledAvatarImage imgAlt={imgAlt} imgSrc={imgSrc}/>)
          : icon
            ? (<Icon name={icon}/>)
            : this.props.children

      }

    </StyledAvatar>)
  }
}

export default Avatar;
