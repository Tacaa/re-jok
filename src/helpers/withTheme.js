import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
