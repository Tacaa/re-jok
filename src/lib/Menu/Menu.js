// @flow
import * as React from 'react';
import {
  StyledMenu
} from './style';

import { isArray, isChild } from '../helpers/typeUtils';


import MenuList from './MenuList';
import MenuItem from './MenuItem';
import SubMenu from './SubMenu';
import {MenuContext} from './MenuContext';

type Props = {
  /** Add style to Menu**/
  style?: Object,
  /** Add more class to Menu**/
  className?:string,
  /** Children of Menu could be anything.. but i recommand API of Menu**/
  children?: any,
  /** Controll Menu.. set background of Menu**/
  backgroundColor?: string,
  /** Controll Menu.. set text color of Menu**/
  textColor?: string,
  /** Controll Menu.. set text color when active of Menu**/
  activeColor?: string,
  /** Controll Menu.. set active color is background or text **/
  activeNormal: boolean,
  /** Callback function when select menu item**/
  onSelect?: Function,
  /** Callback function when deselect menu item**/
  onDeselect?: Function,
  /****/
  activeKey?: string,
  /** Allow Menu is select or not**/
  selectable?: boolean,
  /****/
  selectedKeys?: Array<string>,
  /** Default menu item is selected **/
  defaultSelectedKeys?: Array<string>,
  /** Allow multiple select or not **/
  multiple: boolean,
  /** Default submenu open or not **/
  defaultOpenKeys?: Array<string>,
  /****/
  openKeys?: Array<string>
}

const defaultProps = {
  selectable: true,
  defaultSelectedKeys: [],
  defaultOpenKeys: [],
  multiple: false,
  activeNormal: true
}

type State = {
  selectedKeys: Array<string>,
  openKeys: Array<string>,
  multiple: boolean

}

class Menu extends React.Component<Props, State>{

  static defaultProps = defaultProps;

  static List = MenuList;
  static Item = MenuItem;
  static Sub = SubMenu;

  constructor(props: Object){
    super(props);


    let selectedKeys = props.defaultSelectedKeys;
    let openKeys = props.defaultOpenKeys;

    if ('selectedKeys' in props) {
      selectedKeys = props.selectedKeys || [];
    }

    if ('openKeys' in props) {
      openKeys = props.openKeys || [];
    }

    this.state = {
      selectedKeys,
      openKeys,
      multiple: props.multiple
    }

  }

  onSelect = (itemInfo: Object) => {
    const { selectable, onSelect, multiple } = this.props;

    if (selectable) {
      // root menu
      let { selectedKeys } = this.state;
      const selectedKey = itemInfo.key;

      if (multiple) {
        selectedKeys = [...selectedKeys, selectedKey];
      } else {
        selectedKeys = [selectedKey];
      }
      if (!('selectedKeys' in this.props)) {
        this.setState({
          selectedKeys,
        });
      }
      //onSelect is function user call
      onSelect && onSelect({
        ...itemInfo,
        selectedKeys,
      });
    }
  }

  onOpenChange = (event: Object) => {

    let {openKeys} = this.state;
    let changed = false;
    const processSingle = (e) => {
      let oneChanged = false;
      if (e.open) {
        oneChanged = openKeys.indexOf(e.key) === -1;
        if (oneChanged) {

          openKeys = [...openKeys, e.key];
        }
      } else {
        const index = openKeys.indexOf(e.key);
        oneChanged = index !== -1;
        if (oneChanged) {

          openKeys = openKeys.filter(key => key !== e.key);
        }
      }
      changed = changed || oneChanged;
    };


    isArray(event) ? event.forEach(processSingle) : processSingle(event)

    if (changed) {
      if (!('openKeys' in this.props)) {

        this.setState({ openKeys });
      }
      // props.onOpenChange(openKeys);
    }
  }

  onDeselect = (itemInfo: Object) => {
    const { selectable, onDeselect } = this.props;

    if (selectable) {
      const {selectedKeys} = this.state;
      const selectedKey = itemInfo.key;
      const index = selectedKeys.indexOf(selectedKey);
      if (index !== -1) {
        selectedKeys.splice(index, 1);
      }
      if (!('selectedKeys' in this.props)) {
        this.setState({
          selectedKeys,
        });
      }
      onDeselect && onDeselect({
        ...itemInfo,
        selectedKeys,
      });
    }
  }

  render(){
    const {
      children,
      multiple,
      activeColor,
      activeNormal,
      ...rest
    } = this.props;

    const hasChild = !isChild(children);

    const { selectedKeys, openKeys } = this.state;
    return (
      <MenuContext.Provider value={{
          selectedKeys,
          openKeys,
          onSelect: this.onSelect,
          onDeselect: this.onDeselect,
          onOpenChange:  this.onOpenChange,
          multiple: multiple,
          activeColor: activeColor,
          activeNormal: activeNormal,
        }}>
        <StyledMenu {...rest}>
          {hasChild && children}
        </StyledMenu>
      </MenuContext.Provider>

    )
  }
}

export default Menu;
