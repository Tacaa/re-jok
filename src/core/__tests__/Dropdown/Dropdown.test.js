import React from 'react';
import Dropdown from '../../Dropdown/Dropdown';
import Menu from '../../Menu/Menu';
import Divider from '../../Divider';
import Avatar from '../../Avatar/Avatar';
import { defaultTheme } from '../../theme';

import { shallowWithTheme, mountWithTheme, renderWithTheme } from '../../helpers/withTheme';
import 'jest-enzyme';
import 'jest-styled-components';

const MenuDemo = ()=> (
  <div style={{ width: 256 }}>
    <Menu style={{marginBottom: 20}}>
      <Menu.List title="Header">
        <Menu.Item onClick={(e)=> console.log(e)} keyActive="1" primaryText="Item 1" />
        <Menu.Item onClick={(e)=> console.log(e)} keyActive="2" primaryText="Item 2"/>
        <Menu.Item onClick={(e)=> console.log(e)} keyActive="3" primaryText="Item 3"/>
      </Menu.List>
      <Divider/>
    <Menu.Item onClick={(e)=> console.log(e)} keyActive="4" primaryText="Item 4"/>


    </Menu>
  </div>
)

describe('<Dropdown />', () => {
  it('should render Dropdown', () => {

    const tree = renderWithTheme(<Dropdown content={<MenuDemo/>} position="bottomLeft">
      <Avatar style={{cursor: "pointer"}} imgSrc="https://scontent.fhan2-1.fna.fbcdn.net/v/t1.0-1/c8.0.160.160/p160x160/13256326_1693157114242503_1952158310037985017_n.jpg?_nc_cat=0&_nc_eui2=AeHxGBHGerdp_tpPrKAqCXG_dT5O7zEHBxGSVtiqVcY-NJyK4OXo7i_pH-6207vmY_0yJ4iocMGa3IgmYhgSfCYPuAPyLQQBTnP-vPg-U_wZtw&oh=a421582f63736bf60c96f745a7ef9aa2&oe=5B843852"/>

      </Dropdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });





})
