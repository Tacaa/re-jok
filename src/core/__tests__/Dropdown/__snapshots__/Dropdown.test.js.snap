// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render Dropdown 1`] = `
.c0 {
  position: relative;
}

.c3 {
  max-width: 25.6rem;
  position: absolute;
  display: none;
  top: 0px;
  left: 0px;
  z-index: 500;
  padding: 1rem 0;
  background-color: #fff;
  border: #e8e8e8;
  font-size: 1.4rem;
  border-radius: .4rem;
  box-shadow: 0 .5rem 1.2rem rgba(0,0,0,0.15);
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  outline: none;
  list-style: none;
  background: #fff;
  color: rgba(0,0,0,.64);
  font-size: 1.4rem;
  -webkit-transition: background .3s,width .2s;
  transition: background .3s,width .2s;
  border-right: .1rem solid #e8e8e8;
}

.c4::before {
  content: "";
  display: table;
}

.c4::after {
  clear: both;
  content: "";
  display: table;
}

.c7 {
  cursor: pointer;
  width: 100%;
  position: relative;
  text-overflow: ellipsis;
  line-height: 4rem;
  height: 4rem;
  padding-right: 1.6rem;
  padding-left: 2.4rem;
  box-sizing: border-box;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c7:hover {
  background-color: rgba(200,200,200,0.2);
  outline: none;
  box-shadow: none;
}

.c5 {
  color: rgba(0,0,0,0.48);
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 1.5;
  padding: .8rem 1.6rem;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c5:first-child {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c6 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c8 {
  display: block;
  height: .1rem;
  width: 100%;
  margin: 1.6rem 0;
  clear: both;
  vertical-align: middle;
  position: relative;
  top: -.06em;
  font-size: 1.4rem;
  line-height: 1.5;
  color: rgba(0,0,0,.64);
  box-sizing: border-box;
  list-style: none;
  background: #e8e8e8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style: none;
  text-transform: uppercase;
  box-sizing: border-box;
  margin-right: .8rem;
  background: transparent;
  color: #fafafa;
  font-size: 1.4rem;
  line-height: 1.5;
  width: 3.2rem;
  height: 3.2rem;
  padding: 0;
  z-index: 3;
  border-radius: 50%;
}

.c2 {
  border-radius: inherit;
  width: 100%;
  height: 100%;
  object-fit: cover;
  vertical-align: middle;
  border-style: none;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="c1"
    shape="circle"
    size="default"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <img
      alt=""
      className="c2"
      src="https://scontent.fhan2-1.fna.fbcdn.net/v/t1.0-1/c8.0.160.160/p160x160/13256326_1693157114242503_1952158310037985017_n.jpg?_nc_cat=0&_nc_eui2=AeHxGBHGerdp_tpPrKAqCXG_dT5O7zEHBxGSVtiqVcY-NJyK4OXo7i_pH-6207vmY_0yJ4iocMGa3IgmYhgSfCYPuAPyLQQBTnP-vPg-U_wZtw&oh=a421582f63736bf60c96f745a7ef9aa2&oe=5B843852"
    />
  </div>
  <div
    className="c3"
    open={false}
  >
    <div
      style={
        Object {
          "width": 256,
        }
      }
    >
      <ul
        className="c4"
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <li
          className=""
        >
          <div
            className="c5"
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          >
            Header
          </div>
          <ul
            className="c6"
          >
            <li
              className="c7"
              onClick={[Function]}
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <span
                className=""
              >
                Item 1
              </span>
            </li>
            <li
              className="c7"
              onClick={[Function]}
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <span
                className=""
              >
                Item 2
              </span>
            </li>
            <li
              className="c7"
              onClick={[Function]}
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <span
                className=""
              >
                Item 3
              </span>
            </li>
          </ul>
        </li>
        <div
          className="c8"
          type="horizontal"
        />
        <li
          className="c7"
          onClick={[Function]}
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            className=""
          >
            Item 4
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
